CMakeLists.txt

```diff
@@ -1,53 +1,47 @@
-cmake_minimum_required(VERSION 3.9)
-
-project(VulkanMemoryAllocator LANGUAGES CXX)
-
-# VulkanMemoryAllocator contains an sample application which is not built by default
-option(VMA_BUILD_SAMPLE "Build VulkanMemoryAllocator sample application" OFF)
-option(VMA_BUILD_SAMPLE_SHADERS "Build VulkanMemoryAllocator sample application's shaders" OFF)
-
-message(STATUS "VMA_BUILD_SAMPLE = ${VMA_BUILD_SAMPLE}")
-message(STATUS "VMA_BUILD_SAMPLE_SHADERS = ${VMA_BUILD_SAMPLE_SHADERS}")
-
-option(VMA_STATIC_VULKAN_FUNCTIONS "Link statically with Vulkan API" ON)
-option(VMA_DYNAMIC_VULKAN_FUNCTIONS "Fetch pointers to Vulkan functions internally (no static linking)" OFF)
-option(VMA_DEBUG_ALWAYS_DEDICATED_MEMORY "Every allocation will have its own memory block" OFF)
-option(VMA_DEBUG_INITIALIZE_ALLOCATIONS "Automatically fill new allocations and destroyed allocations with some bit pattern" OFF)
-option(VMA_DEBUG_GLOBAL_MUTEX "Enable single mutex protecting all entry calls to the library" OFF)
-option(VMA_DEBUG_DONT_EXCEED_MAX_MEMORY_ALLOCATION_COUNT "Never exceed VkPhysicalDeviceLimits::maxMemoryAllocationCount and return error" OFF)
-
-message(STATUS "VMA_STATIC_VULKAN_FUNCTIONS = ${VMA_STATIC_VULKAN_FUNCTIONS}")
-message(STATUS "VMA_DYNAMIC_VULKAN_FUNCTIONS = ${VMA_DYNAMIC_VULKAN_FUNCTIONS}")
-message(STATUS "VMA_DEBUG_ALWAYS_DEDICATED_MEMORY = ${VMA_DEBUG_ALWAYS_DEDICATED_MEMORY}")
-message(STATUS "VMA_DEBUG_INITIALIZE_ALLOCATIONS = ${VMA_DEBUG_INITIALIZE_ALLOCATIONS}")
-message(STATUS "VMA_DEBUG_GLOBAL_MUTEX = ${VMA_DEBUG_GLOBAL_MUTEX}")
-message(STATUS "VMA_DEBUG_DONT_EXCEED_MAX_MEMORY_ALLOCATION_COUNT = ${VMA_DEBUG_DONT_EXCEED_MAX_MEMORY_ALLOCATION_COUNT}")
-
-if(VMA_STATIC_VULKAN_FUNCTIONS OR VMA_BUILD_SAMPLE)
-    find_package(Vulkan REQUIRED)
-else()
-    find_package(VulkanHeaders REQUIRED)
-endif()
+cmake_minimum_required(VERSION 3.23...3.26)
```

Comment:
Note: If requested I can lower the CMake version required. FILE_SET was introduced in `3.23` which makes the CMake code a lot easier to read/write for a header only library.

Reply:
Here is what the VulkanHeaders CMake code looks like for reference (Which has 3.15 as the minimum):
https://github.com/KhronosGroup/Vulkan-Headers/blob/main/CMakeLists.txt

Reply:
It's worth pointing out though that the find_package support is backwards compatible:

![image](https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/assets/114601453/3a385270-985c-40b3-bd3f-4b1afdfd8f26)

So vcpkg/conan/etc users won't be affected by the CMake minimum required.